tests/backup/backup_restore_basic_test.go

```diff
@@ -2763,3 +2764,458 @@ var _ = Describe("{BackupRestoreOnDifferentK8sVersions}", func() {
+						validity = false
+						errHelpStr = "Error. help: ensure no remnant cluster-level resources on destination cluster."
+					} else if initialRestoreStatus == api.RestoreInfo_StatusInfo_Success && laterRestoreStatus == api.RestoreInfo_StatusInfo_Success {
+						validity = false
+						errHelpStr = "Error. help: ensure app has cluster-level resources."
+					}
+					dash.VerifyFatal(validity, true, fmt.Sprintf("states of (initial,later) restore are (Success,PartialSuccess) or (PartialSuccess,Success).\n %s", errHelpStr))
+				})
+
+			}
+		})
+
+	})
+
+	JustAfterEach(func() {
+		log.InfoD("Begin JustAfterEach")
+		defer func() { log.InfoD("End JustAfterEach") }()
+
+		defer func() {
+			log.InfoD("switching to default context")
+			err1 := SetClusterContext("")
+			log.FailOnError(err1, "failed to SetClusterContext to default cluster")
+		}()
+
+		defer EndTorpedoTest()
+
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "fetching px-central-admin ctx")
+
+		if len(preRuleNameList) > 0 {
+			for _, ruleName := range preRuleNameList {
+				err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
+				dash.VerifySafely(err, nil, fmt.Sprintf("deleting backup pre rules %s", ruleName))
+			}
+		}
+		if len(postRuleNameList) > 0 {
+			for _, ruleName := range postRuleNameList {
+				err := Inst().Backup.DeleteRuleForBackup(orgID, ruleName)
+				dash.VerifySafely(err, nil, fmt.Sprintf("deleting backup post rules %s", ruleName))
+			}
+		}
+
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = false
+		log.InfoD("deleting deployed applications for source and destination clusters")
+
+		log.InfoD("switching to source context")
+		err = SetSourceKubeConfig()
+		log.FailOnError(err, "failed to switch to context to source cluster")
+
+		log.InfoD("deleting deployed applications on source clusters")
+		ValidateAndDestroy(scheduledAppContexts, opts)
+
+		log.InfoD("waiting (for 1 minute) for any Resources created by Operator of Custom Resources to finish being destroyed.")
+		time.Sleep(time.Minute * 1)
+		log.Warn("no verification of destruction is done; it might lead to undetectable errors")
+
+		// Restored stuff destruction will be added here in upcoming PR
+
+		log.InfoD("switching to default context")
+		err = SetClusterContext("")
+		log.FailOnError(err, "failed to SetClusterContext to default cluster")
+
+		backupDriver := Inst().Backup
+		log.InfoD("deleting backed up namespaces")
+		for _, backupName := range backupNames {
+			backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+			log.FailOnError(err, "failed while trying to get backup UID for - %s", backupName)
+			backupDeleteResponse, err := DeleteBackup(backupName, backupUID, orgID, ctx)
+			log.FailOnError(err, "backup [%s] could not be deleted", backupName)
+			dash.VerifyFatal(backupDeleteResponse.String(), "", fmt.Sprintf("verifying [%s] backup deletion is successful", backupName))
+		}
```

Comment:
I am assuming CleanupCloudSettingsAndClusters will take care of deleting backups.

Reply:
I think it has to be done manually, as shown here.
Does deleting a backup location also delete corresponding backups? If yes, then this code is not needed.